test/integration_test.rb

```diff
@@ -0,0 +1,62 @@
+require 'helpers/kubeclient_helper'
+
+module KubernetesDeploy
+  class IntegrationTest < ::Minitest::Test
+    include KubeclientHelper
+
+    def setup
+      @logger_stream = StringIO.new
+      @logger = Logger.new(@logger_stream)
+      KubernetesDeploy.logger = @logger
+      @namespace = TestProvisioner.claim_namespace
+    end
+
+    def teardown
+      @logger_stream.close
+      TestProvisioner.delete_namespace(@namespace)
+    end
+  end
+
+  module TestProvisioner
+    extend KubeclientHelper
+
+    def self.claim_namespace
```

Comment:
The current test name is a great idea. It looks like [`Minitest::Test#run`](https://github.com/seattlerb/minitest/blob/master/lib/minitest/test.rb#L107) will have the name of the current test set as `self.name`. The whole Minitest stacktrace can be seen in [this comment](https://github.com/seattlerb/minitest/blob/master/lib/minitest.rb#L110). Perhaps `self.name` will do the right thing here? You might have to get a bit more creative with a debugger, but if you dump the methods available with you in `pry` or `byebug` you might be able to figure it out.